src/dmd/opover.d

```diff
@@ -478,9 +478,17 @@ private Expression checkAliasThisForRhs(AggregateDeclaration ad, Scope* sc, BinE
  * Operator overload.
  * Check for operator overload, if so, replace
  * with function call.
- * Return NULL if not an operator overload.
+ * Params:
+ *      e = expression with operator
+ *      sc = context
+ *      pop = if not null, is set to the operator that was actually overloaded,
+ *            which may not be `e.op`. Happens when operands are reversed to
+ *            to match an overload
+ * Returns:
+ *      `null` if not an operator overload,
+ *      otherwise the lowered expression
  */
-Expression op_overload(Expression e, Scope* sc)
+Expression op_overload(Expression e, Scope* sc, TOK* pop = null)
```

Comment:
You're violating some of your own principles here:  prefer `ref` to pointers, avoid default parameters.

Reply:
I know. I did it to minimize the diffs. Also, the new rvalue ref change will support defaulting ref parameters, which will nicely avoid this little bit of ugliness.